pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -670,6 +534,306 @@ def phase_str(phi: float) -> str:
+    draw_phase_curve: bool = False,
+) -> tuple[Figure | None, Figure]:
+    """Draws the entire sequence.
+
+    Args:
+        seq: The input sequence of operations on a device.
+        sampling_rate: Sampling rate of the effective pulse used by
+            the solver. If present, plots the effective pulse alongside the
+            input pulse.
+        draw_phase_area: Whether phase and area values need to be shown
+            as text on the plot, defaults to False. If `draw_phase_curve=True`,
+            phase values are ommited.
+        draw_interp_pts: When the sequence has pulses with waveforms of
+            type InterpolatedWaveform, draws the points of interpolation on
+            top of the respective waveforms (defaults to True).
+        draw_phase_shifts: Whether phase shift and reference information
+            should be added to the plot, defaults to False.
+        draw_register: Whether to draw the register before the pulse
+            sequence, with a visual indication (square halo) around the qubits
+            masked by the SLM, defaults to False.
+        draw_input: Draws the programmed pulses on the channels, defaults
+            to True.
+        draw_modulation: Draws the expected channel output, defaults to
+            False. If the channel does not have a defined 'mod_bandwidth', this
+            is skipped unless 'draw_input=False'.
+        draw_phase_curve: Draws the changes in phase in its own curve (ignored
+            if the phase doesn't change throughout the channel).
+    """
+
+    def phase_str(phi: float) -> str:
+        """Formats a phase value for printing."""
+        value = (((phi + np.pi) % (2 * np.pi)) - np.pi) / np.pi
+        if value == -1:
+            return r"$\pi$"
+        elif value == 0:
+            return "0"  # pragma: no cover - just for safety
+        else:
+            return rf"{value:.2g}$\pi$"
+
+    n_channels = len(seq.declared_channels)
+    if not n_channels:
+        raise RuntimeError("Can't draw an empty sequence.")
+
+    # Sample the sequence and get the data to plot
+    sampled_seq = sample(seq)
+    data = gather_data(sampled_seq)
+
+    # Gather additional data for sequence specific drawing
+    for ch, sch in seq._schedule.items():
+        interp_pts: defaultdict[str, list[list[float]]] = defaultdict(list)
+
+        for slot in sch:
+            if slot.ti == -1 or slot.type in ["target", "delay"]:
+                continue
+
+            pulse = cast(Pulse, slot.type)
+            for wf_type in ["amplitude", "detuning"]:
+                wf = getattr(pulse, wf_type)
+                if isinstance(wf, InterpolatedWaveform):
+                    pts = wf.data_points
+                    pts[:, 0] += slot.ti
+                    interp_pts[wf_type] += pts.tolist()
+
+        if interp_pts:
+            data[ch].interp_pts = dict(interp_pts)
+
+    if hasattr(seq, "_measurement"):
+        data["measurement"] = seq._measurement
```

Comment:
if `hasattr(seq, "_measurement")` then `sampled_seq` has an attribute `_measurement`. I propose you move this to `draw_channel_content`.

Reply:
do you mean to `gather_data` as it was before?

Reply:
Yes indeed !

Reply:
The check becomes `if sampled_seq._measurement is not None`